`virtctl expand --file` should accept `-` (STDIN)

/area instancetype

**Is your feature request related to a problem? Please describe**:
$subject, at present this fails with the following:

```
$ ./cluster-up/kubectl.sh apply -k https://github.com/kubevirt/common-instancetypes.git
[..]
$ ./cluster-up/virtctl.sh create vm --instancetype u1.medium --preference fedora --volume-containerdisk src:quay.io/containerdisks/fedora:40 | ./cluster-up/virtctl.sh expand --file -
error reading file open -: no such file or directory
```

**Describe the solution you'd like**:
Support for `-f -` as with other `kubectl` commands.

**Describe alternatives you've considered**:
N/A

**Additional context**:
N/A

it did not work 
i even delete docker and reinstall it and still get the error
i am using Ubuntu 22.04.4 LTS 
this the error i am getting when i run make cluster-sync
```bash
Error pushing image to localhost:32840/kubevirt/virt-operator:devel: unable to push image to localhost:32840/kubevirt/virt-operator:devel: Get "http://localhost:32840/v2/": dial tcp 127.0.0.1:32840: connect: connection refused
make[1]: *** [Makefile:159: cluster-build] Error 1
make[1]: Leaving directory '/home/ali/Projects/kubevirt'
make: *** [Makefile:168: cluster-sync] Error 2
```
maybe there is some libraries or docker images are missing

--- a/pkg/virtctl/vm/expand.go
+++ b/pkg/virtctl/vm/expand.go
@@ -24,6 +24,7 @@ import (
 	"context"
 	"encoding/json"
 	"fmt"
+	"io"
 	"os"
 
 	"github.com/spf13/cobra"
@@ -84,7 +85,7 @@ func (o *Command) expandRun(args []string, cmd *cobra.Command) error {
 			return fmt.Errorf("error expanding VirtualMachine - %s in namespace - %s: %w", vmName, namespace, err)
 		}
 	} else {
-		vm, err := readVMFromFile(filePath)
+		vm, err := readVMFromFileOrStdin(filePath)
 		if err != nil {
 			return err
 		}
@@ -111,6 +112,9 @@ func usageExpand() string {
   # Expand a virtual machine from file called myvm.yaml.
   {{ProgramName}} expand --file myvm.yaml
 
+  # Expand a virtual machine from stdin. Press Ctrl+D (Unix) or Ctrl+Z (Windows) to signal the end of input.
+  {{ProgramName}} expand --file -
+
   # Expand a virtual machine called myvm and display output in json format.
   {{ProgramName}} expand --vm myvm --output json
   `
@@ -130,16 +134,23 @@ func expandArgs() cobra.PositionalArgs {
 	}
 }
 
-func readVMFromFile(filePath string) (*v1.VirtualMachine, error) {
+func readVMFromFileOrStdin(payload string) (*v1.VirtualMachine, error) {
 	vm := &v1.VirtualMachine{}
 
-	readFile, err := os.ReadFile(filePath)
-	if err != nil {
-		return nil, fmt.Errorf("error reading file %+w", err)
+	var data []byte
+	var err error
+
+	if payload == "-" {
+		if data, err = io.ReadAll(os.Stdin); err != nil {
+			return nil, fmt.Errorf("error reading from stdin: %+w", err)
+		}
+	} else {
+		if data, err = os.ReadFile(payload); err != nil {
+			return nil, fmt.Errorf("error reading file %+w", err)
+		}
 	}
 
-	err = yml.NewYAMLOrJSONDecoder(bytes.NewReader(readFile), 1024).Decode(&vm)
-	if err != nil {
+	if err = yml.NewYAMLOrJSONDecoder(bytes.NewReader(data), 1024).Decode(&vm); err != nil {
 		return nil, fmt.Errorf("error decoding VirtualMachine %+w", err)
 	}
 

--- a/pkg/virtctl/vm/expand_test.go
+++ b/pkg/virtctl/vm/expand_test.go
@@ -48,7 +48,26 @@ var _ = Describe("Expand command", func() {
 	var vmInterface *kubecli.MockVirtualMachineInterface
 	var expandSpecInterface *kubecli.MockExpandSpecInterface
 	var ctrl *gomock.Controller
-	const vmName = "testvm"
+	const (
+		vmName = "testvm"
+		vmSpec = `apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  name: testvm
+spec:
+  runStrategy: Always
+  template:
+    spec:
+      domain:
+        devices: {}
+        machine:
+          type: q35
+        resources: {}
+        volumes:
+status:
+`
+		invalidYaml = `apiVersion: kubevirt.io/v1kind: VirtualMachine`
+	)
 
 	BeforeEach(func() {
 		ctrl = gomock.NewController(GinkgoT())
@@ -107,25 +126,6 @@ var _ = Describe("Expand command", func() {
 	Context("with file input", func() {
 		var file *os.File
 
-		const (
-			vmSpec = `apiVersion: kubevirt.io/v1
-kind: VirtualMachine
-metadata:
-  name: testvm
-spec:
-  runStrategy: Always
-  template:
-    spec:
-      domain:
-        devices: {}
-        machine:
-          type: q35
-        resources: {}
-        volumes:
-status:
-`
-			invalidYaml = `apiVersion: kubevirt.io/v1kind: VirtualMachine`
-		)
 		BeforeEach(func() {
 			var err error
 			file, err = os.CreateTemp(GinkgoT().TempDir(), "file-*")
@@ -164,10 +164,57 @@ status:
 			Expect(yaml.Unmarshal([]byte(invalidYaml), vm)).ToNot(Succeed())
 
 			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", fileInput, file.Name())
-			err := cmd()
 
-			Expect(err).To(HaveOccurred())
-			Expect(err).Should(MatchError("error decoding VirtualMachine error converting YAML to JSON: yaml: mapping values are not allowed in this context"))
+			Expect(cmd()).Should(MatchError("error decoding VirtualMachine error converting YAML to JSON: yaml: mapping values are not allowed in this context"))
+		})
+	})
+
+	Context("with stdin input", func() {
+		var (
+			oldStdin *os.File
+			r        *os.File
+			w        *os.File
+			err      error
+		)
+
+		writeToStdin := func(data string) {
+			go func() {
+				defer w.Close()
+				_, _ = w.Write([]byte(data))
+			}()
+		}
+
+		BeforeEach(func() {
+			oldStdin = os.Stdin
+			r, w, err = os.Pipe()
+			Expect(err).ToNot(HaveOccurred())
+			os.Stdin = r
+			DeferCleanup(func() {
+				os.Stdin = oldStdin
+				r.Close()
+				w.Close()
+			})
+		})
+
+		It("should succeed when called with valid stdin input", func() {
+			Expect(yaml.Unmarshal([]byte(vmSpec), vm)).To(Succeed())
+
+			kubecli.MockKubevirtClientInstance.EXPECT().ExpandSpec(k8smetav1.NamespaceDefault).Return(expandSpecInterface).Times(1)
+			expandSpecInterface.EXPECT().ForVirtualMachine(vm).Return(vm, nil).Times(1)
+
+			writeToStdin(vmSpec)
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", fileInput, "-")
+
+			Expect(cmd()).To(Succeed())
+		})
+
+		It("should fail when called with invalid yaml in stdin input", func() {
+			writeToStdin(invalidYaml)
+
+			cmd := clientcmd.NewRepeatableVirtctlCommand("expand", fileInput, "-")
+
+			Expect(cmd()).To(MatchError("error decoding VirtualMachine error converting YAML to JSON: yaml: mapping values are not allowed in this context"))
 		})
 	})
 })

